File: Linear/Dynamic/Linked List/Circular Linked List/02_Deleting/Beginning/Deleting.java
class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class DeletingBeginning {
    public static Node deleteAtBeginning(Node last) {
        Node head = last.next;

        if (last == null || last == head) {
            return null;
        }

        last.next = head.next;

        head = null;

        return last;
    }

    public static void printList(Node last) {
        if (last == null)
        return;

        Node head = last.next;
        Node temp = head;

        do {
            System.out.print(temp.data);
            temp = temp.next;
            if (temp != head)
            System.out.print(" -> ");
        } while (temp != head);

        System.out.println();
    }

    public static void main(String[] args) {
        // Create circular linked list: 2 -> 3 -> 4
        Node first = new Node(2);
        first.next = new Node(3);
        first.next.next = new Node(4);

        Node last = first.next.next;
        last.next = first;

        System.out.println("Linked List before deletion");
        printList(last);

        last = deleteAtBeginning(last);

        System.out.println("Linked List after deletion");
        printList(last);
    }
}

File: Linear/Dynamic/Linked List/Circular Linked List/02_Deleting/Beginning/Deleting.js
class Node {
    constructor(data) {
        this.data = data;
        this.next = null;
    }
}

function deleteAtBeginning(last) {
    let head = last.next;

    if (last === null || last === head)
        return null;

    last.next = head.next;

    head = null;

    return last;
}

function printList(last) {
    if (last === null)
        return;

    let head = last.next;
    let temp = head;

    do {
        process.stdout.write(temp.data.toString())
        temp = temp.next;
        if (temp !== head)
            process.stdout.write(" -> ");
    } while (temp !== head);

    console.log();
}

// Create circular linked list: 2 -> 3 -> 4
let first = new Node(2);
first.next = new Node(3);
first.next.next = new Node(4);

let last = first.next.next;
last.next = first;

console.log("Linked List before deletion");
printList(last);

last = deleteAtBeginning(last);

console.log("Linked List after deletion");
printList(last);